src/python/grpcio/grpc/_cython/_cygrpc/completion_queue.pyx.pxi

```diff
@@ -12,10 +12,23 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+cimport cpython
+
+import threading
+import time
 
 cdef int _INTERRUPT_CHECK_PERIOD_MS = 200
 
 
+cdef grpc_event _grpc_completion_queue_next(grpc_completion_queue *cq,
+                                            gpr_timespec deadline,
+                                            void *reserved) except *:
+    # Wrap the call to the C++ core in an 'except *' helper to propagate any
+    # errors raised by the gevent poller callback
+    with nogil:
+      return grpc_completion_queue_next(cq, deadline, reserved)
```

Comment:
Do you have the performance impact percentage of this change? Our benchmark framework is a bit hard to use. I would recommend writing a simple client and server to measure it yourself.

Reply:
I created a very simple [test](https://github.com/beandrad/grpc_perf_test); not sure if that's what you are referring to.

Reply:
I played around the client. The QPS is much lower than my expectation, it should be 5k-7k for unary, 1 thread, gevent. Anyway, the old/new result demonstrated a 4% QPS regression (753.2 to 723.3).

Can you explain more about why we need `except *` to catch the exception, instead of using `PyErr_CheckSignals`?

Reply:
Otherwise the loop in [`_next()`](https://github.com/grpc/grpc/blob/99ac3cb03340a5dd4407bd696d6d5176b1d7eac7/src/python/grpcio/grpc/_cython/_cygrpc/completion_queue.pyx.pxi#L39) never breaks. Another solution is https://github.com/grpc/grpc/pull/26058/commits/b82a655e97af08120627f950cbed49c2449b239b, but I'm not sure if it'd have any side effect.